### ISSUE TITLE
.unionAll does not have the same call signature as .union

### ISSUE THREAD

**Issue by schmod** [tgriesser/knex#3054]
# Environment

Knex version: 0.16.3
Database + version: MySQL 5.7
OS: N/A

# Bug

Knex's documentation states that `.unionAll()` has the same call signature as `.union()`.

> unionAll — `.unionAll(query)`
> Creates a union all query, with the same method signature as the union method.

In practice, `.unionAll()` only supports the callback syntax, ie:

```js
knex.select('*').from('users').whereNull('last_name').unionAll(function() {
  this.select('*').from('users').whereNull('first_name');
})
```

However, `.union()` supports two additional call signatures, which are _not_ supported by `.unionAll()`, ie:

```js
knex.select('*').from('users').whereNull('last_name').unionAll([
  knex.raw('select * from users where first_name is null'),
  knex.raw('select * from users where email is null')
])
```

Additionally, `unionAll()` can take the same `wrap` parameter that's supported by `.union()`, but this is not mentioned in the documentation.

**This discrepancy is readily apparent when comparing [the implementations](https://github.com/tgriesser/knex/blob/4db047df02653ec9e70ef004e38f47851231c62d/src/query/builder.js#L609-L643) of `.union()` `.unionAll()`**.

# Error message

tl;dr; it's nothing helpful (which seems like a problem itself, but is probably a topic for a different bug report).  knex tries to invoke every function method on the query builder, and finally throws an exception while trying to start a MySQL transaction.

---

tl;dr; `.unionAll()` should be updated to support the same method signature as `.union()` (there's no reason not to), and the documentation should be updated to show that `.unionAll()` also supports `wrap`
